src/compiler/types.ts

```diff
@@ -1485,11 +1485,11 @@ module ts {
     // Class and interface types (TypeFlags.Class and TypeFlags.Interface)
     export interface InterfaceType extends ObjectType {
         typeParameters: TypeParameter[];           // Type parameters (undefined if non-generic)
-        declaredProperties: Symbol[];              // Declared members
-        declaredCallSignatures: Signature[];       // Declared call signatures
-        declaredConstructSignatures: Signature[];  // Declared construct signatures
-        declaredStringIndexType: Type;             // Declared string index type
-        declaredNumberIndexType: Type;             // Declared numeric index type
+        declaredProperties: Symbol[];              // Declared members (computed by resolveDeclaredMembers)
+        declaredCallSignatures: Signature[];       // Declared call signatures (computed by resolveDeclaredMembers)
+        declaredConstructSignatures: Signature[];  // Declared construct signatures (computed by resolveDeclaredMembers)
+        declaredStringIndexType: Type;             // Declared string index type (computed by resolveDeclaredMembers)
+        declaredNumberIndexType: Type;             // Declared numeric index type (computed by resolveDeclaredMembers)

```

Comment:
If you leave these members in InterfaceType, then someone could access them directly and forget to call resolveDeclaredMembers.

I prefer hiding these in a separate type, so that a cast is necessary to access them. If you really want to access the property (because you are resolveDeclaredMembers and that's your job), then you can cast. Otherwise, you have to call something that calls resolveDeclaredMembers for you.